Sync bundle row state and bundle detail state

leilenah opened the pull request:
### Reasons for making this change

The state shown in the bundle row is currently not synced with the state shown in the expanded bundle details. This change ensures that the bundle row and the expanded bundle details will always be in sync.

### Related issues

https://github.com/codalab/codalab-worksheets/issues/4174

### Screenshots

#### Before (prod)

https://user-images.githubusercontent.com/25855750/188984560-5a1ef10e-a1fc-42d7-b086-ebd5d9046577.mp4

#### After (local)

https://user-images.githubusercontent.com/25855750/188984600-2adff012-ed1c-446a-b1a5-e80a8e4e4c6b.mp4

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [ ] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [ ] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

leilenah committed:
Let bundle row dictate bundle state information.

leilenah committed:
Refactor state info logic.

leilenah committed:
Merge branch 'master' into fix/4174-state-sync

percyliang reviewed (commented):
I think the `stateInfo` terminology might be confusing, because `state` is a specific field within a bundle...

percyliang commented:
Could you describe at a high-level what the fix was?

leilenah committed:
Merge branch 'master' into fix/4174-state-sync

leilenah committed:
Make state info variable names more clear.

leilenah commented:
@percyliang the worksheet view fetches basic bundle info on an interval in order to keep the rows up-to-date. The BundleDetail view also fetches basic bundle info on an interval (as well as more detailed information that the worksheet view doesn't need to fetch).

The issue is that the fetching isn't synced, so the fix I implemented is to allow the worksheet to dictate what state to display in the bundle detail view if necessary. 

Note that I've updated some `stateInfo` variable names to make things more clear.

percyliang reviewed (commented):
Got it.  Would you mind updating the code with a comment that describes how the bundle information gets propagated from the table to the bundle detail page to keep things synchronized?

leilenah committed:
Merge branch 'master' into fix/4174-state-sync

leilenah committed:
Refactor bundle state info logic.

leilenah committed:
Merge branch 'fix/4174-state-sync' of https://github.com/codalab/codalab-worksheets into fix/4174-state-sync

leilenah committed:
Revert detail sidebar change.

leilenah commented:
@percyliang refactored a bit. lmk if this works

percyliang reviewed (approved):
A few minor comments, but overall looks good to me - thanks!

leilenah committed:
Merge branch 'master' into fix/4174-state-sync

leilenah committed:
Update renderDuration helper.
